src/helpers/messaging.rs

```diff
@@ -207,6 +294,18 @@ impl Gateway {
                         #[cfg(debug_assertions)]
                         print_state(role, &send_buf, &receive_buf);
                     }
+                    Ok(()) = &mut shutdown_rx => {
```

Comment:
Can you explain why you see a need for an explicit shutdown?  Do we have a problem with circular references, such that this sort of cleanup is necessary?

Reply:
It's not functionally needed, but it makes it possible to check that all channels have been cleaned up at the end of tests.